src/main/java/org/example/commands/DeleteDependencyCommand.java

```diff
@@ -0,0 +1,40 @@
+package org.example.commands;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Scanner;
+import org.example.exceptions.InvalidArgument;
+import org.example.model.Node;
+
+public class DeleteDependencyCommand implements CommandInterface {
+
+  @Override
+  public void execute(Map<String, Node> nodeDependencies) {
+    Scanner scanner = new Scanner(System.in);
+    System.out.println("Get parent id and child id");
+
+    String parentId = scanner.nextLine();
+    String childId = scanner.nextLine();
+
+    // Check if both parent and child nodes exist in the map
+    if (nodeDependencies.containsKey(parentId) && nodeDependencies.containsKey(childId)) {
+
+      // Update the parent node: remove the child id
+      ArrayList<String> children =
```

Comment:
follow defensive coding methods. If nodeDependencies does not consist parentId then getNodeChildren will throw null pointer exception. Instead first check nodeDependencies.containsKey(parentId)